src/Core/Checkout/Promotion/Subscriber/PromotionIndividualCodeRedeemer.php

```diff
@@ -128,7 +128,7 @@ private function collectLineItems(EntityWrittenEvent $event): OrderLineItemColle
         $orderLineItems = new OrderLineItemCollection();
 
         foreach ($event->getWriteResults() as $result) {
-            if ($result->getPayload()['type'] !== PromotionProcessor::LINE_ITEM_TYPE) {
+            if (!isset($result->getPayload()['type']) || $result->getPayload()['type'] !== PromotionProcessor::LINE_ITEM_TYPE) {
```

Comment:
```suggestion
            if (($result->getPayload()['type'] ?? '') !== PromotionProcessor::LINE_ITEM_TYPE) {
```

I also fixed that in my PR, but it is not merged yet :sweat_smile: with the null coalescing operator you can omit the isset